File: 21551-naveen-siva-kumar/week10/122_Best Time to Buy and Sell Stock II.py
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int sum=0,diff=0;
        for(int i=1;i<prices.size();i++){
            diff=prices[i]-prices[i-1];
            if(diff>0)
                sum+=diff;
        }
        return sum;
        
    }
};

File: 21561 - Suvam Tamang/Week 10/Best Time to Buy and Sell Stock II/122. Best Time to Buy and Sell Stock II.java
class Solution {
    public int maxProfit(int[] prices) {
        // Simple O(n) solution, Greedy Approach!

        // int sum = 0;
        // for(int i=0;i<prices.length-1;i++)
        // {
        //     if(prices[i+1]>prices[i])
        //         sum = sum + prices[i+1]-prices[i];
        // }
        // return sum;

        int n = prices.length;
        int[] dp = new int[n];
        dp[0]=0;
        for(int i=1;i<n;i++){
            dp[i] = dp[i-1] + Math.max(0,prices[i]-prices[i-1]);
        }
        return dp[n-1];
    }
}
